pkg/spoke/managedcluster/creating_controller.go

```diff
@@ -53,14 +53,31 @@ func NewManagedClusterCreatingController(
 }
 
 func (c *managedClusterCreatingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	_, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
+	var managedClusterClientConfigs []clusterv1.ClientConfig
+	if len(c.spokeExternalServerURLs) != 0 {
+		for _, serverURL := range c.spokeExternalServerURLs {
+			managedClusterClientConfigs = append(managedClusterClientConfigs, clusterv1.ClientConfig{
+				URL:      serverURL,
+				CABundle: c.spokeCABundle,
+			})
+		}
+	}
+
+	existingCluster, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
 	switch {
 	case errors.IsUnauthorized(err),
 		errors.IsForbidden(err) && strings.Contains(err.Error(), anonymous):
 		klog.V(4).Infof("unable to get the managed cluster %q from hub: %v", c.clusterName, err)
 		return nil
 	case errors.IsNotFound(err):
 	case err == nil:
+		if len(existingCluster.Spec.ManagedClusterClientConfigs) == 0 && len(managedClusterClientConfigs) > 0 {
+			clusterCopy := existingCluster.DeepCopy()
+			clusterCopy.Spec.ManagedClusterClientConfigs = managedClusterClientConfigs
+			if _, err := c.hubClusterClient.ClusterV1().ManagedClusters().Update(ctx, clusterCopy, metav1.UpdateOptions{}); err != nil {
+				return fmt.Errorf("unable to update ManagedClusterClientConfigs of managed cluster %q on hub: %w", c.clusterName, err)
```

Comment:
I do not think the agent has the permission to update this field. By default, it should not.